### PR TITLE
Fixed warnings

### PR DESCRIPTION
**Pull request by vijetaR**
### Checklist:
- [ ] My code is tested
- [ ] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: admin/class-hfe-admin.php [hunk 1]
```diff
@@ -448,8 +449,8 @@ public function ehf_save_meta( $post_id ) {
 			return;
 		}
 
-		$target_locations = array_map( 'sanitize_text_field', Astra_Target_Rules_Fields::get_format_rule_value( $_POST, 'bsf-target-rules-location' ) );
-		$target_exclusion = array_map( 'sanitize_text_field', Astra_Target_Rules_Fields::get_format_rule_value( $_POST, 'bsf-target-rules-exclusion' ) );
+		$target_locations = Astra_Target_Rules_Fields::get_format_rule_value( $_POST, 'bsf-target-rules-location' );
+		$target_exclusion = Astra_Target_Rules_Fields::get_format_rule_value( $_POST, 'bsf-target-rules-exclusion' );
```
[the remark this answers is not in the document]
Comment (vijetaR) [reply admin/class-hfe-admin.php:453]: Checking for if `is_array` then **_array map_** is not allowing to save the input fields for Display condition and throws warning further. I check for alternative, not giving expected results.

File: admin/class-hfe-admin.php [hunk 2]
```diff
@@ -194,9 +194,11 @@ public function column_content( $column, $post_id ) {
 	public function column_display_location_rules( $locations ) {
 
 		$location_label = [];
-		$index          = array_search( 'specifics', $locations['rule'] );
-		if ( false !== $index && ! empty( $index ) ) {
-			unset( $locations['rule'][ $index ] );
+		if ( isset( $locations['rule'] ) && is_array( $locations['rule'] ) ) {
```
Comment (sushmak02) [inline admin/class-hfe-admin.php:197]: is_array( $locations )

File: admin/class-hfe-admin.php [hunk 3]
```diff
@@ -194,9 +194,11 @@ public function column_content( $column, $post_id ) {
 	public function column_display_location_rules( $locations ) {
 
 		$location_label = [];
-		$index          = array_search( 'specifics', $locations['rule'] );
-		if ( false !== $index && ! empty( $index ) ) {
-			unset( $locations['rule'][ $index ] );
+		if ( isset( $locations['rule'] ) && is_array( $locations ) ) {
```
Comment (sushmak02) [inline admin/class-hfe-admin.php:197]: if ( is_array( $locations ) && isset( $locations['rule'] ) ) {

File: admin/class-hfe-admin.php [hunk 4]
```diff
@@ -194,9 +194,11 @@ public function column_content( $column, $post_id ) {
...
+			$index = array_search( 'specifics', $locations['rule'] );
+			if ( false !== $index && ! empty( $index ) ) {
```
Comment (sushmak02) [inline admin/class-hfe-admin.php:199]: && is_array( $locations['rule'] )

File: inc/lib/target-rule/class-astra-target-rules-fields.php [hunk 5]
```diff
@@ -765,72 +765,73 @@ public static function generate_target_rule_selector( $type, $selection_options,
 		}
 
 		$index = 0;
+		if ( is_array( $saved_values['rule'] ) ) {
```
Comment (sushmak02) [inline inc/lib/target-rule/class-astra-target-rules-fields.php:768]: if ( is_array( $saved_values ) && is_array( $saved_values['rule'] ) ) {

**Review by sushmak02 — APPROVED** [review]

**Pull request merged**